test/jdk/java/awt/PrintJob/PrintCheckboxTest/PrintCheckboxManualTest.java

```diff
@@ -24,82 +24,73 @@
-//  harness will not mistake this file as a test file.  It should
-//  only see the html file as a test file. (the harness runs all
-//  valid test files, so it would run this test twice if this file
-//  were valid as well as the html file.)
-// Also, note the area= after Your Name in the author tag.  Here, you
-//  should put which functional area the test falls in.  See the
-//  AWT-core home page -> test areas and/or -> AWT team  for a list of
-//  areas.
-
-
-
-import java.awt.*;
-import java.awt.event.*;
-
-
-//Manual tests should run as applet tests if possible because they
-// get their environments cleaned up, including AWT threads, any
-// test created threads, and any system resources used by the test
-// such as file descriptors.  (This is normally not a problem as
-// main tests usually run in a separate VM, however on some platforms
-// such as the Mac, separate VMs are not possible and non-applet
-// tests will cause problems).  Also, you don't have to worry about
-// synchronisation stuff in Applet tests the way you do in main
-// tests...
-
-
-public class PrintCheckboxManualTest extends Panel
+import java.awt.Button;
+import java.awt.Checkbox;
+import java.awt.EventQueue;
+import java.awt.Frame;
+import java.awt.Graphics;
+import java.awt.GridLayout;
+import java.awt.PrintJob;
+import java.awt.Scrollbar;
+import java.awt.Toolkit;
+import java.awt.print.PrinterJob;
+import java.io.IOException;
+import java.lang.reflect.InvocationTargetException;
+import java.util.function.Consumer;
+import java.util.function.Supplier;
+import javax.swing.JEditorPane;
+import jtreg.SkippedException;
+import lib.ManualTestFrame;
+import lib.TestResult;
+
+public class PrintCheckboxManualTest
 {
-    //Declare things used in the test, like buttons and labels here
-    Frame f;
 
-    public static void main(String[] args) {
-        PrintCheckboxManualTest a = new PrintCheckboxManualTest();
-
-        a.init();
-        a.start();
-    }
-
-    public void init()
-    {
-        //Create instructions for the user here, as well as set up
-        // the environment -- set the layout manager, add buttons,
-        // etc.
-        this.setLayout (new BorderLayout ());
+    public static void main(String[] args) throws InterruptedException,
+            InvocationTargetException, IOException {
+        if (PrinterJob.lookupPrintServices().length == 0) {
+            throw new SkippedException("Printer not configured or available."
+                    + " Test cannot continue.");
+        }
 
-        String[] instructions =
-        {
-            "Linux or Solaris with XToolkit ONLY!",
-            "1. Click the 'Print' button on the frame",
-            "2. Select a printer in the print dialog and proceed",
-            "3. If the frame with checkbox and button on it is printed successfully test PASSED else FAILED"
+        String instruction = """
+               Linux or Solaris with XToolkit ONLY!
```

Comment:
since there already is
@requires (os.family == "linux" | os.family == "solaris")
is this line needed? 
Can there be anything else besides XToolkit on linux or solaris? If yes, is there a way to determine that programmatically? If no, how does user finds that out?
Lastly, what is the test executor supposed to do if it is not XToolkit on linux or solaris?